Season-2/Level-3/code.py

```diff
@@ -40,37 +39,16 @@
 def index():
     if request.method == 'POST':
         planet = request.form.get('planet')
-        sanitized_planet = re.sub(r'[<>(){}[\]]', '', planet)
-
-        if 'script' in sanitized_planet.lower() :
-            return '<h2>Blocked</h2></p>'
-
-        elif sanitized_planet:
-            details = get_planet_info(sanitized_planet)
-
-            if planet:
-                return f'<h2>Planet Details:</h2><p>{get_planet_info(planet)}</p>'
-            else:
-                return '<h2>Please enter a planet name.</h2>'
-
-    return render_template('index.html')
-
-@app.route('/getPlanetInfo', methods=['GET'])
-def get_planet_info_endpoint():
-    planet = request.args.get('planet')
-    sanitized_planet = re.sub(r'[<>(){}[\]]', '', planet)
```

Comment:
@evarga The idea behind this level and especially this specific line of code was to show that many devs might reinvent the wheel and try to blocklist some symbols to avoid injection attacks. Therefore, I feel that the despite the code could be simplified more of course, the goal was to trick students to think that is not vulnerable due to the checks implemented. What do you think?

Reply:
The `sanitized_planet` variable contains a decently escaped input, which means none of the provided examples of tainted payloads would work. For example, the original regular expression removes all occurrences of `<` and `>` from the input. The solution text was written under an assumption that this will not happen at all, i.e., that `<<` and `>>` would remain intact.

This is why I have recommended to skip sanitization in the current form; otherwise, students would need to apply a dozen of advanced tricks to figure out how to inject improper payload despite the above sanitization.

Reply:
We need students to face some difficulty in order to learn more in this level and I see your point about the solution assumption, so if possible, feel free to propose something on the solution file and keep the rest of structure (and regex in code file) as it is, or with minimal changes to keep the essence of the game. 

The learning objective was to show students an example of reinventing the wheel through block-listing symbols instead of using a library etc. It's a good idea to show them that when a developer creates a manual control, it's under the assumption that inputs will be X, Y, Z, while in reality, skilled attackers can do W, Q, etc

Reply:
I agree that an additional level of complexity will definitely improve the learning experience. I will insert back the checks into the `index` function to simulate a false sense of security without overcomplicating the mechanism of injecting tainted payload into a system. Should be ready this week.

Reply:
Thank you!!!